pyoptsparse/pyOpt_utils.py

```diff
@@ -570,3 +574,38 @@ def _broadcast_to_array(name: str, value: ArrayType, n_values: int, allow_none:
     if not allow_none and any([i is None for i in value]):
         raise Error(f"The {name} argument cannot be 'None'.")
     return value
+
+
+def try_import_compiled_module_from_path(module_name: str, path: Optional[str] = None) -> Union[types.ModuleType, str]:
+    """
+    Attempt to import a module from a given path.
+
+    Parameters
+    ----------
+    module_name : str
+        The name of the module
+    path : Optional[str]
+        The path to import from. If None, the default ``sys.path`` is used.
+
+    Returns
+    -------
+    Union[types.ModuleType, str]
+        If importable, the imported module is returned.
+        If not importable, the error message is instead returned.
+    """
+    orig_path = sys.path
+    if path is not None:
+        path = os.path.abspath(os.path.expandvars(os.path.expanduser(path)))
+        sys.path = [path]
+    try:
+        module = importlib.import_module(module_name)
+    except ImportError as e:
+        if path is not None:
+            warnings.warn(
+                f"{module_name} module could not be imported from {path}.",
+                stacklevel=2,
+            )
+        module = str(e)
+    finally:
+        sys.path = orig_path
+    return module
```

Comment:
Is there a reason why we delay throwing the `ImportError`? If we can't import the module, why not throw the error with the descriptive message here instead of a warning?  

Do we need this "soft fail" type behavior somewhere that I missed?  Otherwise I'm in favor of just throwing the `ImportError` here with the same message and only returning a module if the import is successful.  But again, let me know if I missed something that makes this necessary.

Reply:
Yeah it's because this is executed as part of the init process, and we don't necessarily want to raise an error. But if they try to instantiate the class and the compiled library is missing, we throw the error.

Reply:
Okay that makes sense.  I forgot that python tries to import anything that's involved in the init process.  I'll approve this now.